### ISSUE TITLE
Poe-Harvest-Mechanics-Comments

### ISSUE THREAD

**Issue by ForgottenArbiter** [ForgottenArbiter/forgottenarbiter.github.io#4]

**Comment by masonk** [comment]
Just wanted to say thank you. The data was still useful for me to plan crop rotation.


Here is the expected value of a plot after it has been upgraded by crop rotation n times.

<img width="3559" height="2964" alt="Image" src="https://github.com/user-attachments/assets/6606eb88-bcb4-4c76-b2f2-0e8d185023c7" />

**Comment by masonk** [comment]
I also wrote a solver that finds the provably optimal (highest EV) path through any harvest, if you are interested in that I can share it with you, although, it rarely plays the game differently than an experienced Exile would.

**Comment by elementalshu** [comment]
> I also wrote a solver that finds the provably optimal (highest EV) path through any harvest, if you are interested in that I can share it with you, although, it rarely plays the game differently than an experienced Exile would.

Any way I could look at this? I would like to check it out. :)

**Comment by ForgottenArbiter** [comment]
@masonk nice to see people doing analysis on crop rotation! I have seen a few other people making attempts to analyze crop rotation. It's an interesting problem and the decision tree is large enough that determining an exact optimal strategy isn't so trivial. A lot of the analysis has focused on trying to figure out what distribution of colors to use in crop rotation. Interestingly, everybody seems to get slightly different answers, though everybody has found that minimizing the weight of yellow (assuming yellow is the most valuable color) is a bad strategy. A lot of the differences in results could easily come down to differences in assumptions. 

Here are some of the projects I've seen (but I haven't really given any of them more than a cursory look):

* https://github.com/JZongit/Crop-Rotation-Simulator
* https://www.youtube.com/watch?v=gmONlZN5hqU

There are some more in the Prohibited Library discord. Maybe @elementalshu would be interested too.

**Comment by masonk** [comment]
Yes, I went further and solved the optimal atlas passives assuming optimal play. Unlike earlier efforts, I am not doing monte carlo simulation, but analytically solved it (with the aid of a computer).

However the analysis is only valid assuming optimal play, and optimal play requires imputing monetary value to the seed counts. So the analysis is only valid for prices approximately like what they were in 3.26.


https://docs.google.com/spreadsheets/d/1zIGVDc3JEJ-YD6iPIHr6RCD0SCG8S0UAZVvjYkIfGLU/edit

I used combinatorics to calculate the odds of receiving each distinct kind of initial plot, under the assumption that it is 25:50:25 likelihood of 3:4:5 plots. Then for each kind of starting plot I used an analytic solver to pick the best path through the harvest and count the expected number of each kind of seed.

The code is in Rust as I found python too slow. It can calculate optimal play for even the worst 5-plot harvests in <2s.

**Comment by masonk** [comment]
Anyway, I found that the optimal strategy from a div/harvest perspective is taking everything except one yellow small passive. However the yield curve was very flat and ultimately I decided that I could get more $ by spending 6 passives elsewhere.

**Comment by masonk** [comment]
> > I also wrote a solver that finds the provably optimal (highest EV) path through any harvest, if you are interested in that I can share it with you, although, it rarely plays the game differently than an experienced Exile would.
> 
> Any way I could look at this? I would like to check it out. :)



I’ll upload the code one day. I am just working on the mathematical writeup that goes with it…

**Comment by masonk** [comment]
Ok, here we go: https://github.com/masonk/cropbot?tab=readme-ov-file

**Comment by mofa007** [comment]
This is an absolutely fantastic and deeply insightful piece of work. Thank you for sharing such a comprehensive, data-driven analysis of Harvest mechanics—it's an incredible resource.

Inspired by your post, I developed a fully automated Crop Rotation simulator. Your work provided the essential underlying probabilities, and my simulator aims to answer a subsequent question: given the current dynamic market prices, which Atlas passive configuration yields the highest expected profit?

Due to the inherent randomness, the approach is to run tens of thousands of simulations and then analyze the results statistically (mean, median, mode) to derive a more robust "optimal" passive tree.

Please note:​ I used AI to help me write this comment and the simulator's logic, so I apologize for any grammatical errors or mistakes. I am a player from China, so the simulator's interface is in Chinese. You could likely use AI or other tools to translate it.

I would be very grateful if you could take a look at the simulator's content, particularly the part about my harvesting strategy. I had a quick look at your post, and my understanding is that your strategy might be fixed at the start. In contrast, my simulator's sequence is determined after each harvest: it calculates the outcome based on the probabilities you discovered and then decides the next best plot to harvest dynamically.

Finally, a few details about this simplified simulator:

It models the standard 8-plot Sacred Grove (not 6 or 10-plot layouts from memories).

It does not factor in IIQ, pack size, or Atlas passives (like the "chance for an additional monster").

It only simulates up to Tier 3 seeds, and only T3 seeds are considered to yield a "harvest" (i.e., lifeforce).

I also built a non-Crop-Rotation simulator (to decide when to spec into the keystone—I found it's optimal when the price of Yellow lifeforce is greater than the sum of Blue and Purple) and a manual Crop Rotation simulator (now temporarily lost, possibly on some cloud service).

Thanks again for this stellar contribution.

[harvest_simulator.html](https://github.com/user-attachments/files/24388594/harvest_simulator.html)
[harvest_simulator.py](https://github.com/user-attachments/files/24388593/harvest_simulator.py)

**Comment by masonk** [comment]
In Cropbot’s codebase, I don’t have any simulator, but I do provide an answer for optimal atlas configuration (number of reduced purple-blue-yellow nodes to take to maximize profit under a fixed market condition).

I do this by calculating the optimal strategy for every starting plot, the probability of each starting plot, and the expected number of seeds you get from each starting plot under optimal play. The only thing that changes under different atlas configs is the probability of each starting plot, so it’s easy to recompute the weighted sums under different configurations.

So there is no simulation, but I am giving something better: the mathematically proven optimal configuration, with no randomness that could possibly mess up the results or call them into question.

**Comment by marjotoska** [comment]
Hi, I find these 2 sections contradicting:

"Quantity from eldritch altars affects amount of lifeforce dropped
I ran a 0% quantity map, took a significant amount of quantity from eldritch altars, and the stack sizes of dropped lifeforce were inconsistent with 0% area IIQ and consistent with the amount of area IIQ obtained from the altars. The quantity from altars also directly adds to the area IIQ displayed in-map."

and

"Player IIQ does not affect amount of lifeforce dropped
Several players have dropped lifeforce with -100% player IIQ (which normally suppresses all non-deterministic item drops). I also recorded the amount of player IIQ when collecting my own data, and could not find any correlation between my player IIQ and amount of lifeforce dropped."

Due to my knowledge, one can only increase item quantity via "Player Gains" choices from Eldritch Altars, e.g.: "Player gains 16% increased item quantity of items found in this area". 

So my question is: does the following upside affect the quantity of lifeforce dropped, or not? Thanks :)

**Comment by masonk** [comment]
Eldritch altars do affect because they change the tooltip on the map effect. Player iiq doesn’t affect because it doesn’t change the map effect stat.

I know the altar says player gains but it actually adds to the map effect iiq stat. Easy to verify by looking at the map mods overlay before and after you click the altarS
